pandas/core/arrays/_ranges.py

```diff
@@ -46,8 +46,8 @@ def generate_regular_range(
     ndarray[np.int64]
         Representing the given resolution.
     """
-    istart = start.value if start is not None else None
-    iend = end.value if end is not None else None
+    istart = start._value if start is not None else None
```

Comment:
im a bit concerned about this.  we have a unit keyword that i think is implicitly assuming unit == start.unit == end.unit

Reply:
no wait, never mind, this is doing the right thing given that concern.